lib/node_modules/@stdlib/repl/lib/validate.js

```diff
@@ -107,6 +107,12 @@ function validate( opts, options ) {
 			return new TypeError( format( 'invalid option. `%s` option must be a string. Option: `%s`.', 'welcome', options.welcome ) );
 		}
 	}
+	if ( hasOwnProp( options, 'themes' ) ) {
```

Comment:
This isn't the only option which needs validating. Also, what should happen if `theme` doesn't correspond to a list of supported themes, either provided or built-in?

Reply:
I assume you are asking what happens if a user provides invalid themes? The user won't see the colors.

Reply:
I assume you are asking what happens if a user provides invalid themes? The user won't see the colors.

Reply:
Two things: we still need to validate a provided `theme` option, as the `themes` option is not the only one which needs to be validated. For example, `options.theme` should be a string.

And a user not seeing themes is not great UX. Is that because they provided invalid input (e.g., a theme name which doesn't exist in the table of themes) or is it because we have a bug? A user won't be able to discern that and will wonder why expected colors are not showing. In general, we should fail as early as we can.

Reply:
@kgryte The `theme` option is passed as a setting (ie. `options.settings.theme`) and is being validated by the settings validator. And if a user gives a theme that doesn't exist in the table of themes, it will throw an error telling the user to give an existing theme as input.
When I said the user won't see the colors, I meant it for when user gives their own table of themes using `options.themes` and as we have already documented the tokens we support, I don't think the user would wonder why the colors aren't showing.

Reply:
Providing invalid tokens is still something a user would likely want to know. If they provide an invalid token and don't see colors, that seems like a rather indirect way of informing them that they made a mistake. In general, this should be addressed through a JSON schema and validator. So I'll mark as resolved, but I suggest you open an RFC issue so we don't forget that this task still needs completing.